web/packages/teleport/src/Login/useLogin.ts

```diff
@@ -39,7 +39,16 @@ export default function useLogin() {
   const auth2faType = cfg.getAuth2faType();
   const isLocalAuthEnabled = cfg.getLocalAuthFlag();
   const motd = cfg.getMotd();
-  const [showMotd, setShowMotd] = useState(!!motd);
+  const [showMotd, setShowMotd] = useState(() => {
+    const url = new URL(window.location.href);
```

Comment:
Nit: I think it is better to use `react-router` APIs, instead of accessing and operating on `window` directly. 
```tsx
  const location = useLocation();
  const [showMotd, setShowMotd] = useState(() => {
    const searchParams = new URLSearchParams(location.search);
    const redirectUri = searchParams.get('redirect_uri');
...
```

Reply:
Following your suggestion, I changed the code to use `useLocation`. But then found it actually adds a dependency to all the tests used in `Login.test.tsx: Jest: TypeError: Cannot read properties of undefined (reading 'location')`. That seems to be solved by wrapping the test component to render inside react router (e.g. `<MemoryRouter><Login /><MemoryRouter>`). But the fact that I will have to update other tests which depends on location: https://github.com/gravitational/teleport/blob/884bd7fb3847c8173c50a3f28cbfe1fa7e625e66/web/packages/teleport/src/Login/Login.test.tsx#L27-L31 I am cautious if this change would add any extra value given the risk of bugging existing test logic (tests are passing locally after wrapping inside `<MemoryRouter />` but still...)

What do you think? I recon it would be following recommended practice to use `react-router` but wanted to check on your opinion here.

Reply:
Ah, probably it doesn't make sense to add another dependency there. 
However, I think we could use `history.getLocation()`, at lest we will relay on a single API.

But if this won't work, I'm fine with `window.location`

Reply:
`history.getLocation()` had the same effect, but `history.getRedirectParam()` worked like charm. Used that instead of direct `window.location` in https://github.com/gravitational/teleport/pull/31050/commits/73b6f2e93082206a12a0b90415c3d11719ef0551

Thanks for pointing to that.